record-editor/src/app/core/services/record-cleanup.service.ts

```diff
@@ -61,4 +67,16 @@ export class RecordCleanupService {
     // boolean, number
     return false;
   }
+
+  public isURL(value: any): boolean {
+    if (typeof value !== 'string') {
+      return false;
+    }
+    const pattern = /(http|https):\/\/(\w+:{0,1}\w*)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%!\-\/]))?/;
```

Comment:
you could use https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
Maybe I'm missing sometihng, but isn't the whole point that you want to encode things that are not valid URLs? What if there's a ` ` or a :smiley: in the value, but is otherwise a URL? will your regex detect that? conversely, what if an abstract contains a URL? are you now going to encode the full abstract?

I don't know if it's easy to do, but it seems more robust to do this based on the `format` expected by the schema, instead of the value and try to guess whether it should be a URL.